src/plugins/expressions/common/expression_types/specs/datatable.ts

```diff
@@ -127,13 +148,13 @@ export const datatable: ExpressionTypeDefinition<typeof name, Datatable, Seriali
     }),
     pointseries: (table: Datatable): PointSeries => {
       const validFields = ['x', 'y', 'color', 'size', 'text'];
-      const columns = table.columns.filter((column) => validFields.includes(column.name));
+      const columns = table.columns.filter((column) => validFields.includes(column.id));
```

Comment:
Are changes like these going to break existing Canvas workpads where someone may have constructed or transformed their own datatable?

In other words, can we populate `id` with `name` if `id` doesn't exist?  Or is that already handled I just don't see it...?

Reply:
we are setting `id` to `name` everywhere, so this should still work.

Reply:
For aggs the plan was to keep the `id`s with the deterministic tabify naming scheme like `col-${colIndex}-${aggId}`

For non-esaggs IDs, should we be doing `col-${colIndex}`, or do you think it matters? Trying to think if there'd be a use case for this, but I think we're probably fine as-is. Wanted to double-check to make sure I'm not missing anything though.

Reply:
i would go without this for now, as i am not sure if there are any benefits to it and it would certainly mean larger refactoring